docs/DeveloperGuide.md

```diff
@@ -1024,15 +1024,196 @@ testers are expected to do more *exploratory* testing.
+
+   3. Test case: `remove x` (where x is a number larger than the number of assignments in the assignment list)<br>
+      Expected: No assignment is removed. Error details shown in the status message since the index is invalid.
+
+   4. Test case: `remove 0`<br>
+      Expected: No assignment is removed. Error details shown in the status message since the index is invalid.
+
+2. Removing an assignment while assignment list panel is empty.
+
+   1. Prerequisites: There are multiple persons in the contact list and no person's assignment list is shown.
+   
+   2. Test case: `remove 1`<br>
+      Expected: No assignment is removed. Error details shown in the status message since the index is invalid.
+ 
+3. Removing an assignment while missing compulsory fields.
+
+   1. Test case:`remove`<br>
+      Expected: No assignment is removed from the assignment list panel. Error details shown in the status message since the format is invalid.
+   
+###  Redoing a command
+   
+1. Redoing a command at the start of program.
+
+   1. Test case: `redo`<br>
+      Expected: Nothing is redone. Error details shown in the status message since no state can be redone.
+   
+2. Redoing a command after a `undo` command.
+
+   1. Test case: `redo`<br>
+      Expected: Recovers the effect of last `undo` command.
+   
+3. Redoing a command after a command except `undo`. 
+
+   1. Test case: `redo`<br>
+      Expected: Nothing is redone. Error details shown in the status message since no state can be redone.
+   
+### Undoing a command
+
+1. Undoing a command at the start of program.
+
+   1. Test case: `undo`<br>
+      Expected: Nothing is undone. Error details shown in the status message since no state can be undone.
+
+2. Undoing a command after entering some commands.
+
+   1. Test case: `undo`<br>
+      Expected: Retrieves the effect before conducting the last command (except `undo`).
+   
+### Editing a person
+
+1. Editing a person while all persons are being shown.
+
+   1. Prerequisites: There are multiple persons in the contact list (no person called "Alex" and no email named "15434@163.com").
+
+   2. Test case: `edit 1 n/Alex e/15434@163.com m/CS2100`<br>
+      Expected: The first person in the contact list is renamed as "Alex", and the email and module of this person changed accordingly to the given. 
+      Details of the edited person will be shown in the status message.
+   
+   3. Test case: `edit 2 n/Alex e/11465434@163.com m/CS2100` just after last test case.<br>
+      Expected: No person is edited. Error details shown in the status message since the repeated name is not allowed.
+   
+   4. Test case: `edit 2 n/Alex Yeoh e/15434@163.com m/CS2100` just after last test case.<br>
+      Expected: No person is edited. Error details shown in the status message since the repeated email is not allowed.
+   
+   5. Test case: `edit 2 t/friend t/lab7`<br>
+      Expected: The first person's tag is replaced with tags called "friend" and "lab7".
```

Comment:
The command is `edit 2 ...` but you mentioned first person when it should be second.

Reply:
good catch